tests/testthat/test-derive.R

```diff
@@ -1,8 +1,52 @@
 context("Derive a new variable")
 
+with_mock_crunch({
+    ds <- loadDataset("test ds")
+    # manually make a variable from json mock
+    birthyrPlus <- CrunchVariable(
+        VariableTuple(index_url = "https://app.crunch.io/api/datasets/1/variables/?relative=on",
+                      entity_url = "https://app.crunch.io/api/datasets/1/variables/birthyrPlus/",
+                      body = crGET("https://app.crunch.io/api/datasets/1/variables/birthyrPlus/")$body)
+    )
+
+    test_that("derivation pulls the derivation", {
+        expect_is(derivation(birthyrPlus), "CrunchExpr")
+        expect_true(is.derived(birthyrPlus))
+        expect_fixed_output(derivation(birthyrPlus), "Crunch expression: birthyr + 100")
+    })
+
+    test_that("derivation returns NULL if the variable is not derived", {
+        expect_null(derivation(ds$birthyr))
+    })
+
+    test_that("derivation()<- input validation", {
+        expect_error(derivation(ds$birthyr) <- ds$birthyr + 1,
+                     paste0("The variable ", dQuote("Birth Year"),
+                            " must already be a derived variable."))
+        expect_error(derivation(birthyrPlus) <- "not an expression",
+                     paste0("The value ", dQuote("not an expression"),
+                            " must be a CrunchExpr, got a character instead."))
+    })
+
+    test_that("derivation()<- issues a PATCH",{
+        expect_PATCH(derivation(birthyrPlus) <- ds$birthyr + 1,
+                     'https://app.crunch.io/api/datasets/1/variables/birthyrPlus/',
+                     '{"derivation":{"function":"+","args":[{"variable":"https://app.crunch.io/api/datasets/1/variables/birthyr/"},{"value":1}]}}')
+    })
+
+    test_that("derivation()<-NULL and is.derived()<-FALSE issues a PATCH",{
+        expect_PATCH(derivation(birthyrPlus) <- NULL,
+                     'https://app.crunch.io/api/datasets/1/variables/birthyrPlus/',
+                     '{"derived":false}')
+        expect_PATCH(is.derived(birthyrPlus) <- FALSE,
+                     'https://app.crunch.io/api/datasets/1/variables/birthyrPlus/',
+                     '{"derived":false}')
+    })
```

Comment:
`expect_no_request` on the ones that shouldn't PATCH (e.g. if setting TRUE on a derived variable, if setting FALSE on a non-derived variable, if setting NULL on a non-derived variable)

and what happens if you try to set `is.derived(var) <- TRUE` on a variable that is not derived? that should error